geopandas/array.py

```diff
@@ -424,87 +476,86 @@ def __setstate__(self, state):
 
     @property
     def is_valid(self):
-        return vectorized.is_valid(self._data)
+        return shapely.is_valid(self._data)
 
     @property
     def is_empty(self):
-        return vectorized.is_empty(self._data)
+        return shapely.is_empty(self._data)
 
     @property
     def is_simple(self):
-        return vectorized.is_simple(self._data)
+        return shapely.is_simple(self._data)
 
     @property
     def is_ring(self):
-        return vectorized.is_ring(self._data)
+        return shapely.is_ring(self._data)
 
     @property
     def is_closed(self):
-        return vectorized.is_closed(self._data)
+        return shapely.is_closed(self._data)
 
     @property
     def has_z(self):
-        return vectorized.has_z(self._data)
+        return shapely.has_z(self._data)
 
     @property
     def geom_type(self):
-        return vectorized.geom_type(self._data)
+        res = shapely.get_type_id(self._data)
+        return geometry_type_values[np.searchsorted(geometry_type_ids, res)]
```

Comment:
We still don't know how the spherely implementation will look like. It may be via a `GeographyArray` which would be separate from `GeometryArray`. Unsure yet. I guess we can keep it as is and potentially refactor if that is needed.